mitosheet/mitosheet/step_performers/graph_steps/plotly_express_graphs.py

```diff
@@ -192,56 +199,128 @@ def graph_creation_code(
 
 
 def graph_styling(
-    fig: go.Figure, graph_type: str, column_headers: List[ColumnHeader], filtered: bool
+    fig: go.Figure, graph_type: str, column_headers: List[ColumnHeader], filtered: bool, graph_styling_params: Dict[str, Any]
 ) -> go.Figure:
     """
     Styles the Plotly express graph figure
     """
 
-    # Create the parameters that we use to style the graph
-    graph_title = get_graph_title(column_headers, [], filtered, graph_type)
+    # Create the parameters that we use to construct the graph
+    all_params: Dict[str, Union[ColumnHeader, List[ColumnHeader], str, None, dict]] = dict()
+
+    # Create the graph title param
+    if graph_styling_params['title']['visible']:
+        use_custom_title = 'title' in graph_styling_params['title']
+        if use_custom_title:
+            all_params['title'] = graph_styling_params['title']['title']
+        else:
+            all_params['title'] = get_graph_title(column_headers, [], filtered, graph_type)
+
+    # Create the x axis param
+    if graph_styling_params['xaxis']['visible']:
+        # If the x axis title is visible, then either dispaly the user's custom title or Ploty's default title.
+        use_custom_xaxis_title = 'title' in graph_styling_params['xaxis']
+        # Only apply the xaxis_title if it is set because if we set it to None, then we don't get ploty's default values
+        if use_custom_xaxis_title:
+            all_params['xaxis_title'] = graph_styling_params['xaxis']['title']
+    else: 
+        # Plotly makes us explicitly handle setting the xaxis_title and yaxis_title to None
+        all_params['xaxis_title'] = None
+
+    # Create the y axis title param
+    if graph_styling_params['yaxis']['visible']:
+        # If the y axis title is visible, then either dispaly the user's custom title or Ploty's default title.
+        use_custom_yaxis_title = 'title' in graph_styling_params['yaxis']
+        # Only apply the xaxis_title if it is set because if we set it to None, then we don't get ploty's default values
+        if use_custom_yaxis_title:
+            all_params['yaxis_title'] = graph_styling_params['yaxis']['title']
+    else: 
+        # Plotly makes us explicitly handle setting the xaxis_title and yaxis_title to None
+        all_params['yaxis_title'] = None
+
+    # Create the barmode param
     barmode = get_barmode(graph_type)
+    if barmode is not None:
+        all_params['barmode'] = get_barmode(graph_type)
+
+    # Create the showlegend param
+    all_params['showlegend'] = graph_styling_params['showlegend']
+
+    # Create the range slider param
```

Comment:
Can we move this up to the x-axis title area; just so they are all in the same place? Both in the generate graph code and transpile code, to keep them symmetric!